### PR TITLE
Move assessments out of api_controller

### PR DESCRIPTION
**Pull request by caleybrock**
## What's in this PR?
Going for incremental improvements on the assessments APIs after getting some good feedback in this PR: https://github.com/code-dot-org/code-dot-org/pull/22772

This one moves the methods out of `api_controller` into a new `assessments_controller` and creates a resource for assessments. I've updated the front end with the necessary changes. These changes were suggested here: https://github.com/code-dot-org/code-dot-org/pull/22772#discussion_r192167340

I've also updated the comments for both of these methods as suggestion here: https://github.com/code-dot-org/code-dot-org/pull/22772#discussion_r192164972

## What's not in this PR?
I didn't change any of the actual method code at all (I know there are lots of improvements to make here, but this PR is just organizational).

## What's next?
Move the bulk of the logic in these methods to a model (maybe `level_source`) or just in a separate module in `lib` and add tests for it.

### CODE REVIEW

File: dashboard/config/routes.rb [hunk 1]
```diff
@@ -95,6 +95,15 @@ module OPS
     end
   end
 
+  # Used in react assessments tab
```
Comment (caleybrock) [inline dashboard/config/routes.rb:98]: I'd love feedback about these routes changes in particular. This works, but I haven't written a lot of this code so I'm not sure if this is a reasonable way to go about things.

File: dashboard/app/controllers/api/v1/assessments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    render json: responses_by_student
+  end
+
+  private
+
+  def load_section
+    section = Section.find(params[:section_id])
+    authorize! :read, section
+    section
+  end
```
Comment (aoby) [inline dashboard/app/controllers/api/v1/assessments_controller.rb:190]: When it comes time to refactor, this looks like a perfect candidate for CanCan's `load_and_authorize_resource`

File: dashboard/config/routes.rb (hunk 1 above)
Comment (aoby) [reply dashboard/config/routes.rb:98]: This looks great :)

Granted if it's only used in one place it doesn't need to be specified in a concern, though also there's no harm, just an extra level of indirection. Presumably we'll want to expose this in `/api/v1` as well, at which point the concern will be handy?

File: dashboard/app/controllers/api/v1/assessments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,199 @@
+class Api::V1::AssessmentsController < Api::V1::JsonApiController
+  include LevelsHelper
+
+  # For each assessment in a script, return an object of script_level IDs to question data.
```
Comment (maddiedierker) [inline dashboard/app/controllers/api/v1/assessments_controller.rb:4]: these are great comments! 👍

**Pull request merged**